Zend/zend_compile.c

```diff
@@ -6427,6 +6427,53 @@ static bool can_match_use_jumptable(zend_ast_list *arms) {
 	return 1;
 }
 
+static void zend_compile_pipe(znode *result, zend_ast *ast)
+{
+	zend_ast *operand_ast = ast->child[0];
+	zend_ast *callable_ast = ast->child[1];
+
+	/* Compile the left hand side down to a value first. */
+	znode operand_result;
+	zend_compile_expr(&operand_result, operand_ast);
+
+	/* Wrap the value in a ZEND_QM_ASSIGN opcode to ensure references
+	 * always fail. Otherwise, they'd only fail in complex cases like arrays.
+	 */
+	znode wrapped_operand_result;
+	zend_emit_op_tmp(&wrapped_operand_result, ZEND_QM_ASSIGN, &operand_result, NULL);
```

Comment:
I think it might be safe to elide the QM_ASSIGN if the operand is known not to be a CV, since to my understanding temporary would already result in the desired semantics.

Reply:
It could also be a `VAR`, which may also contain references. Eliding it is guaranteed to be safe for `TMPVAR` and `CONST`. As for the rest, the optimizer will take care of those.

Reply:
I'm afraid I don't understand what Ilija just said.

Reply:
You can check if `operand_result.op_type & (IS_CV|IS_VAR)` and only do `ZEND_QM_ASSIGN` then. Otherwise, you can just assign `operand_result` to `wrapped_operand_result`.